src/SB/Core/x/xString.cpp

```diff
@@ -47,6 +48,84 @@ U32 xStrHashCat(U32 prefix, const char* str)
     return hash;
 }
 
+char* xStrTok(char* string, const char* control, char** nextoken)
+{
+    U8* str;
+    U8* ctrl;
+    U8 map[32];
+    S32 count;
+
+    map[0] = 0;
```

Comment:
We should be able to put this in a more concise for loop that unrolls - please try it out.